pkg/collector/corechecks/containers/generic/processor_test.go

```diff
@@ -34,58 +34,122 @@ func TestProcessorRunFullStatsLinux(t *testing.T) {
+			mockSender.AssertMetric(t, "Rate", "container.cpu.throttled", 100, "", expectedTags)
+			mockSender.AssertMetric(t, "Rate", "container.cpu.throttled.periods", 0, "", expectedTags)
+			mockSender.AssertMetric(t, "Rate", "container.cpu.partial_stall", 96000, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.cpu.limit", 500000000, "", expectedTags)
+
+			mockSender.AssertMetric(t, "Gauge", "container.memory.usage", 42000, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.kernel", 40, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.limit", 42000, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.soft_limit", 40000, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.rss", 300, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.cache", 200, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.working_set", 350, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.swap", 0, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.oom_events", 10, "", expectedTags)
+			mockSender.AssertMetric(t, "Gauge", "container.memory.usage.peak", 50000, "", expectedTags)
+			mockSender.AssertMetric(t, "Rate", "container.memory.partial_stall", 97000, "", expectedTags)
+
+			mockSender.AssertMetric(t, "MonotonicCount", "container.memory.page_faults", 97001, "", expectedTags)
+			mockSender.AssertMetric(t, "MonotonicCount", "container.memory.major_page_faults", 50002, "", expectedTags)
+
+			if tt.extendedMetrics {
+				mockSender.AssertMetric(t, "Gauge", "container.memory.page_tables", 661, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.active_anon", 662, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.inactive_anon", 663, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.active_file", 664, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.inactive_file", 665, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.unevictable", 666, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.shmem", 1481, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.file_mapped", 1482, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.file_dirty", 1483, "", expectedTags)
+				mockSender.AssertMetric(t, "Gauge", "container.memory.file_writeback", 1484, "", expectedTags)
+				mockSender.AssertMetric(t, "MonotonicCount", "container.memory.refault_anon", 1485, "", expectedTags)
+				mockSender.AssertMetric(t, "MonotonicCount", "container.memory.refault_file", 1486, "", expectedTags)
+			} else { // if extendedMetrics processor should not emit those metrics
+				mockSender.AssertMetricMissing(t, "Gauge", "container.memory.page_tables")
```

Comment:
Nit about style, you can simplify this a bit by defining the metrics in an array:

```go
metrics := []struct {
	metricType string
	metricName string
}{
	{"Gauge", "container.memory.page_tables"},
	{"Gauge", "container.memory.active_anon"},

         ...
}

for _, m := range metrics {
	mockSender.AssertMetricMissing(t, m.metricType, m.metricName)
}
```

Reply:
good point. fixed